viewer-prototype/src/node/trace-server-service.ts

```diff
@@ -0,0 +1,14 @@
+import { spawn, exec } from 'child_process';
+import { injectable } from 'inversify';
+import { TraceServerConfigService } from '../common/trace-server-config';
+@injectable()
+export class TraceServerServiceImpl implements TraceServerConfigService {
+
+    async startTraceServer(path: string | undefined, port: number | undefined): Promise<void> {
+        spawn(`${path}`, ['-vmargs', `-Dtraceserver.port=${port}`]);
+    }
+
+    async stopTraceServer(port: number | undefined): Promise<void> {
+        exec(`kill -9 $(lsof -t -i:${port}) -sTCP:LISTEN`);  // FIXME: Better approach to kill the server at a given port
```

Comment:
The -sTCP:LISTEN goes in the parenthesis ;-) My browser still dies.

Reply:
Yes :) Will make that change. It doesn't kill the browser for me. I am using Firefox version 83 and chrome version 75. Could you please share what browser/version you are using?

Reply:
firefox 83 also, with ubuntu beaver. Maybe it's because I usually happen to have a tab opened to some app at port localhost:8080?

Also, does your patch work on Windows? Not that I care ;-) But just for info. What will be Windows users' experience with the trace server?

Reply:
> firefox 83 also, with ubuntu beaver. Maybe it's because I usually happen to have a tab opened to some app at port localhost:8080?
> Yeah could be. Maybe change the URL to something else instead of localhost and test it again? This will also verify if my changes are not breaking the remote case. :)

@Bernd: Are you also facing the same issue?


> Also, does your patch work on Windows? Not that I care ;-) But just for info. What will be Windows users' experience with the trace server?
The last time I played with the window's trace server executable (many months ago) I was running into some path issues. I have not yet played with the latest windows server. I will try and let you know.

Reply:
Fixed the parenthesis.